ospfd/ospf_vty.c

```diff
@@ -3543,6 +3543,27 @@ static void show_ip_ospf_interface_sub(struct vty *vty, struct ospf *ospf,
 				vty_out(vty,
 					"  No backup designated router on this network\n");
 		} else {
+			nbr = ospf_nbr_lookup_by_addr(oi->nbrs, &DR(oi));
+			if (nbr) {
+				if (use_json) {
+					json_object_string_add(
+						json_interface_sub, "DRId",
+						inet_ntoa(nbr->router_id));
+					json_object_string_add(
+						json_interface_sub, "DRaddress",
+						inet_ntoa(nbr->address.u
+								  .prefix4));
+				} else {
+					vty_out(vty,
+						"  Designated Router (ID) %pI4",
+						&nbr->router_id);
+					vty_out(vty,
+						" Interface Address %pFX\n",
+						&nbr->address);
+				}
+			}
+			nbr = NULL;
```

Comment:
Why are we nulling this out here and then repeating the lookup on the next line?

Reply:
Nevermind, misread